lib/src/components/mixins/hitbox.dart

```diff
@@ -1,84 +1,70 @@
-import 'dart:ui';
+import 'dart:collection';
+import 'dart:math';
 
-import '../../../components.dart';
-import '../../collision_detection.dart' as collision_detection;
+import '../position_component.dart';
+import '../../../extensions.dart';
+import '../../geometry/shape.dart';
 
 mixin Hitbox on PositionComponent {
-  List<Vector2> _shape;
+  final List<HitboxShape> _shapes = <HitboxShape>[];
 
-  /// The list of vertices used for collision detection and to define whether
-  /// a point is inside of the component or not, so that the tap detection etc
-  /// can be more accurately performed.
-  /// The hitbox is defined from the center of the component and with
-  /// percentages of the size of the component.
-  /// Example: [[1.0, 0.0], [0.0, 1.0], [-1.0, 0.0], [0.0, -1.0]]
-  /// This will form a square with a 45 degree angle (pi/4 rad) within the
-  /// bounding size box.
-  set shape(List<Vector2> vertices) => _shape = vertices;
-  List<Vector2> get shape => _shape ?? [];
+  UnmodifiableListView<HitboxShape> get shapes => UnmodifiableListView(_shapes);
 
-  /// Whether the hitbox shape has defined vertices and is not an empty list
-  bool hasShape() => _shape?.isNotEmpty ?? false;
+  void addShape(HitboxShape shape) {
+    shape.component = this;
```

Comment:
if shape.component != null you can only use shapes once

Reply:
A HitboxShape should only be used once, since it is bound to the properties of the component, like position, size and angle.

Reply:
sorry I meant `if shape.component != null then throw 'you can only use shapes once'`

Reply:
can we add this check?